climate.py

```diff
@@ -179,15 +203,25 @@ def update(self):
         This is the only method that should fetch new data for Home Assistant.
         """
         state = self._nobo.get_current_zone_mode(self._id, dt_util.as_local(dt_util.now()))
+        self._current_mode = HVAC_MODE_AUTO
+        if state == self._nobo.API.NAME_OFF:
+            self._current_mode = HVAC_MODE_OFF
+            self._current_operation = PRESET_AWAY
+        elif state == self._nobo.API.NAME_AWAY:
+            self._current_operation = PRESET_AWAY
+        elif state == self._nobo.API.NAME_ECO:
+            self._current_operation = PRESET_ECO
+        elif state == self._nobo.API.NAME_COMFORT:
+            self._current_operation = PRESET_COMFORT
         if self._nobo.zones[self._id]['override_allowed'] == '1':
-            self._current_operation = 'Normal (' + state + ')'
             for o in self._nobo.overrides:
                 if self._nobo.overrides[o]['mode'] == '0':
                     continue  # "normal" overrides
                 elif self._nobo.overrides[o]['target_type'] == self._nobo.API.OVERRIDE_TARGET_ZONE:
                     if self._nobo.overrides[o]['target_id'] == self._id:
-                        self._current_operation = state
-        else:
-            self._current_operation = 'Locked (' + state + ')'
+                        self._current_mode = HVAC_MODE_HEAT
+        self._current_temperature = self._nobo.get_current_zone_temperature(self._id)
+        if self._current_temperature == 'N/A':
```

Comment:
I guess we can improve the pynobo function to just return None?

Reply:
Yes, I wanted that, but hesitated to change pynobo